internal/indexlib/bluge/directory/oss/oss.go

```diff
@@ -95,22 +98,67 @@ func ListObjects(
 	return objects, nil
 }
 
-func GetObject(client *oss.Client, bucketName, path string) (io.ReadCloser, error) {
+func GetObject(client *oss.Client, bucketName, path string) ([]byte, error) {
 	bucket, err := GetBucket(client, bucketName)
 	if err != nil {
 		return nil, err
 	}
-	reader, err := bucket.GetObject(path)
+
+	objMeta, err := bucket.GetObjectMeta(path)
+	if err != nil {
+		return nil, err
+	}
+	contentLength := objMeta.Get("Content-Length")
+	size, err := strconv.Atoi(contentLength)
 	if err != nil {
-		logger.Error(
-			"[oss] get object fail",
-			zap.String("bucket", bucket.BucketName),
-			zap.String("path", path),
-			zap.Error(err),
-		)
 		return nil, err
 	}
-	return reader, nil
+
+	var content []byte
+	content = make([]byte, size)
+	gCount := int(math.Min(float64(runtime.GOMAXPROCS(0)), float64(size)))
+	partSize := math.Ceil(float64(size) / float64(gCount))
+
+	var eg errgroup.Group
+	eg.SetLimit(gCount)
+	for i := 0; i < gCount; i++ {
+		start := int64(partSize) * int64(i)
+		end := int64(math.Min(partSize*(float64(i)+1), float64(size)))
+		if start >= int64(size) {
+			break
+		}
+		eg.Go(func() error {
+			object, err := bucket.GetObject(path, oss.Range(start, end-1))
+			if err != nil {
+				logger.Error(
```

Comment:
Printing out the range params of failed fetch requests is handy for troubleshooting.